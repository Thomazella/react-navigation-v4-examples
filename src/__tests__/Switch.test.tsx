/**
 * @format
 */

import 'react-native';
import React from 'react';
import { View, Text } from 'react-native';
import { createSwitchNavigator } from 'react-navigation';

import { renderWithNavigation } from './testUtils';

const AnotherScreen = () => {
  return (
    <View>
      <Text>Another Screen</Text>
    </View>
  );
};

const TestScreen = () => {
  return (
    <View>
      <Text>Test Screen</Text>
    </View>
  );
};

describe('SwitchNavigator', () => {
  it('should render properly', () => {
    const SwitchNavigator = createSwitchNavigator({
      Home: TestScreen,
      Settings: AnotherScreen,
    });
    const rendered = renderWithNavigation(SwitchNavigator);
    expect(rendered.asJSON()).toMatchSnapshot();
  });
});
