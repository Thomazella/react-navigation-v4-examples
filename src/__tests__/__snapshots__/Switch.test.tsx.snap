// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchNavigator should render properly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text>
      Test Screen
    </Text>
  </View>
</View>
`;
